rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+| base file + 2 log files written sequentially -> snapshot read                                                                                         | reads 2 log files in order and merges w/ base file records                                                                     | reads 2 log files in order and merges w/ base file records                                                                               |
+| base file + 2 log files written concurrently w/ OCC and so one of the writer aborted (rollback yet to kick in) -> snapshot read                       | reads just 1 log file and 1 base file, as uncommitted log files will be filtered out.                                          | reads just 1 log file and 1 base file, as uncommitted log files will be filtered out.                                                    |
+| base file + 2 log files written concurrently w/ NBCC and based on completion time log file 2 completed earlier compared to log file1 -> snapshot read | merges all log records. log file2 followed by log file 1 is the order.                                                         | merges all log records. log file1 will be followed by log file 1. We anyway do not support this in 0.16.x reader                         |
+| FG1 having 3rd file slice in pending compaction and few log files added to latest file slice                                                          | latest file slice should be merged w/ log files from last but one file slice. log files are ordered based on completion times. | latest file slice should be merged w/ log files from last but one file slice. log files are ordered based on log version and write token |                                                                |
+| Time travel Query                                                                                                                                     | works as expected                                                                                                              | Assuming commit time is in active timeline, we are good. Else, LSM timeline read support needs to be ported over.                        |
+| Incremental Query                                                                                                                                     | works as expected                                                                                                              | Assuming commit time is in active timeline, we are good. Else, LSM timeline read support needs to be ported over.                        |                                                                                       |
```

Comment:
We are good for 0.16.x if we port the LSM timeline reader and the completion time query view.

Reply:
yes. to support NBCC, we need LSM timeline reader and completion time query view.